CONTRIBUTING.md

```diff
@@ -5,6 +5,76 @@ Welcome, and thank you for your interest in contributing to iTwin.js!
+    - [A subdirectory can not find a node\_modules file or directory](#a-subdirectory-can-not-find-a-node_modules-file-or-directory)
+    - [Updating dependencies/devDependencies on packages within the monorepo](#updating-dependenciesdevdependencies-on-packages-within-the-monorepo)
+    - [Updating dependencies/devDependencies on packages external to monorepo](#updating-dependenciesdevdependencies-on-packages-external-to-monorepo)
+
+## Repo Setup
+
+This repository is a [monorepo](https://en.wikipedia.org/wiki/Monorepo) that holds the source code to multiple iTwin.js npm packages. It is built using [Rush](http://rushjs.io/).
+
+Each package has its own **node_modules** directory that contains symbolic links to *common* dependencies managed by Rush.
+
+1. Install dependencies: `rush install`
+2. Clean: `rush clean`
+3. Build source: `rush build`
+4. Run tests: `rush cover`
+
+The `-v` option for `rush` is short for `--verbose` which results in a more verbose command.
+
+The above commands iterate and perform their action against each package in the monorepo.
+
+## Source Code Edit Workflow
+Once you've set up the repo, you can move on to making changes to the source code by following the steps below:
+1. At the root of the repo, run `git checkout -b "<your-branch-name>"`
+2. Make source code changes
+3. Rebuild the repo by running `rush build`
+4. Ensure unit tests pass when run locally: `rush cover`
+5. Ensure linting passes when run locally: `rush lint`
+6. Locally commit changes: `git commit` (or use the Visual Studio Code user interface)
+7. Repeat steps 1-4 until ready to push changes
+8. Check for API signature changes: `rush extract-api`. This will update the signature files, located in `common/api`. **Note:** before doing this, first do the following:
+    - Be sure that your branch is up to date with the target branch (i.e. `git merge origin/master`)
+    - Cleanup your build output: `rush clean`
+    - Rebuild the project: `rush build`
+9. Review any diffs to the API signature files in the `common/api` directory to ensure they are compatible with the intended release of the package.
+    - If any differences are in packages not modified on this branch, revert the changes before committing.
+10. Add changelog entry (which could potentially cover several commits): `rush change`
+11. Follow prompts to enter a change description or press ENTER if the change does not warrant a changelog entry. If multiple packages have changed, multiple sets of prompts will be presented. If the changes are only to non-published packages (like **display-test-app**), then `rush change` will indicate that a changelog entry is not needed.
+12. Completing the `rush change` prompts will cause new changelog entry JSON files to be created.
+13. Add and commit the changelog JSON files and any API signature updates.
+14. Publish changes on the branch and open a pull request.
```

Comment:
I feel that since all items in this list are complete sentences, they should have periods at the end. If others disagree about this, then they should at least be consistent. (Right now, a few have periods, but most don't.)